ports-of-call/portable_arrays.hpp

```diff
@@ -26,73 +26,145 @@
 
 #include "portability.hpp"
 #include <algorithm>
+#include <array>
 #include <assert.h>
 #include <cstddef> // size_t
 #include <cstring> // memset()
+#include <functional>
+#include <numeric>
+#include <type_traits>
 #include <utility> // swap()
 
+// maximum number of dimensions
+constexpr std::size_t MAXDIM = 6;
+// array type of dimensions/strides
+using narr = std::array<std::size_t, MAXDIM>;
+
+namespace detail {
```

Comment:
It doesn't matter very much, by why `detail` over `impl`?

Reply:
Just habit